crates/re_sdk/src/spawn.rs

```diff
@@ -156,34 +177,76 @@ pub fn spawn(opts: &SpawnOptions) -> Result<(), SpawnError> {
         return Ok(());
     }
 
-    let res = Command::new(executable_path)
+    fn check<T>(opts: &SpawnOptions, res: Result<T, std::io::Error>) -> Result<T, SpawnError> {
+        match res {
+            output @ Ok(_) => output.map_err(Into::into),
+            Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+                if let Some(executable_path) = opts.executable_path.as_ref() {
+                    Err(SpawnError::ExecutableNotFound {
+                        executable_path: executable_path.clone(),
+                    })
+                } else {
+                    Err(SpawnError::ExecutableNotFoundInPath {
+                        message: MSG_INSTALL_HOW_TO.to_owned(),
+                        executable_name: opts.executable_name.clone(),
+                        search_path: std::env::var("PATH").unwrap_or_else(|_| String::new()),
+                    })
+                }
+            }
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    let viewer_version = {
+        let output = check(
+            opts,
+            Command::new(&executable_path).arg("--version").output(),
+        )?;
+
+        // <name> <semver> [<rust_info>] <target> <branch> <commit> <build_date>
+        let output = String::from_utf8_lossy(&output.stdout);
+        let output = output.split_whitespace().collect::<Vec<_>>();
+
+        (output.len() >= 2)
+            .then(|| re_build_info::CrateVersion::try_parse(output[1]).ok())
+            .flatten()
```

Comment:
Please add cross-references to and from `impl std::fmt::Display for BuildInfo` so that we are less likely to break this!

Reply:
Or even better; move this code to that file, i.e. a `fn parse_crate_version_from_build_info_string`. It would also make it trivial to add a minimal roundtrip unit-test